### DISCUSSION TITLE
Using Cloak.nvim

### DISCUSSION THREAD

**Discussion by Eshanatnight** [NvChad/NvChad#2640] [Q&A] [answered]
Hi,
I am trying to add [cloak](https://github.com/laytan/cloak.nvim) to my plugins list.

I have added the plugin in my `plugins.lua`

```lua
    {
        "laytan/cloak.nvim",
        lazy = false,
        opts = function()
            return require "custom.configs.cloak"
        end,

        config = function(_, opts)
            print "loading cloak"
            require("cloak").setup(opts)
        end,
    },
```

and my `customs.configs.cloak.lua` has the basic config from the cloak docs

```lua
local options = {
    enabled = true,
    cloak_character = "*",
    cloak_length = nil,
    try_all_patterns = true,
    patterns = {
        -- Match any file starting with '.env'.
        -- This can be a table to match multiple file patterns.
        file_pattern = ".env*",

        -- Match an equals sign and any character after it.
        -- This can also be a table of patterns to cloak,
        -- example: cloak_pattern = { ':.+', '-.+' } for yaml files.
        cloak_pattern = "=.+",
        -- A function, table or string to generate the replacement.
        -- The actual replacement will contain the 'cloak_character'
        -- where it doesn't cover the original text.
        -- If left emtpy the legacy behavior of keeping the first character is retained.
        replace = nil,
    },
}

return options

```

Now when I am opening any `.env` file the plugin is not activating

Would love some help on this thanks!

**Comment by siduck** [comment]
@Eshanatnight are     you still using nvchad v2.0? always  check lazy.nvim to see if plugin's loaded

**Reply by Eshanatnight** [reply]
Hi @siduck did Lazy sync a couple of weeks ago. I found out that, in the 
`config` function was not getting the opts value. so setup was not happening. I got the same issue when I am using none-ls. 
I required the opts in the config function and pass it to the setup function it works without an issue.

Probably I set something up wrong?

**Reply by siduck** [reply, marked as answer]
@Eshanatnight you dont have to use opts here, we use it so users can override it from their configs. just directly pass in config function

```lua
 {
        "laytan/cloak.nvim",
         event = 'VeryLazy',
        config = function()
            print "loading cloak"
             local opts = require "custom.configs.cloak"
            require("cloak").setup(opts)
        end,
    },

**Reply by Eshanatnight** [reply]
Okay! Thanks for the help with my question!
